### PR TITLE
allow defining docker repository

### PR DESCRIPTION
**Pull request by ReSearchITEng**
due to docker.io (docker hub) limits, setting the repo to a mirror is a must...

### CODE REVIEW

File: templates/calico.yaml [hunk 1]
```diff
@@ -383,7 +383,7 @@ spec:
       securityContext:
         fsGroup: 65534
       containers:
-      - image: docker.io/calico/typha:v{{ .Chart.AppVersion }}
+      - image: {{ .Values.typha.image.repository | default "docker.io" }}/calico/typha:v{{ .Chart.AppVersion }}
```
Comment (invidian) [inline templates/calico.yaml]: I see that Bitnami charts are doing the pattern with `registry`/`repository`/`tag`: https://github.com/bitnami/charts/blob/4ba7e11323e1f787262236cc6707f2f44793d618/bitnami/redis/templates/_helpers.tpl#L67-L88

Perhaps we could do similar, so name a parameter `registry`, we can skip `repository`, `tag` and helper template for now, until it's needed.

Comment (invidian) [reply templates/calico.yaml]: As a side note, if `values.yaml` have default already specified, I think `default "docker.io"` can be skipped here.

**Review by invidian — CHANGES_REQUESTED** [review]
Thanks for the PR @ReSearchITEng. Changes looks good to me, I'd just adjust the naming so we follow some standards. What do you think?

File: templates/calico.yaml (hunk 1 above)
Comment (ReSearchITEng) [reply templates/calico.yaml]: resolved.
Just note, my question is: how many different registries should we have?
Now I've set one for typha, and the one from the node is used in all the other 5 places. I guess one for each section is too much :)

**Comment by ReSearchITEng** [comment]
Just note, my question is: how many different registries should we have?
Now I've set one for typha, and the one from the node is used in all the other 5 places. I guess one for each section is too much :)

File: templates/calico.yaml (hunk 1 above)
Comment (invidian) [reply templates/calico.yaml]: Yeah, I think for now it's fine, thanks!

**Comment by invidian** [comment]
>Just note, my question is: how many different registries should we have?
Now I've set one for typha, and the one from the node is used in all the other 5 places. I guess one for each section is too much :)

I think it's OK as it is for now. It's simple, fits current needs and is extendable in a non-breaking way.

**Review by invidian — CHANGES_REQUESTED** [review]
Just last nit, could you please squash the commits?

**Comment by ReSearchITEng** [comment]
> 
> 
> Just last nit, could you please squash the commits?

Could you use the squash & merge option when you do the merge on your end?

**Comment by invidian** [comment]
>Could you use the squash & merge option when you do the merge on your end?

I could, though I usually don't do that.

**Comment by ReSearchITEng** [comment]
> 
> 
> > Could you use the squash & merge option when you do the merge on your end?
> 
> I could, though I usually don't do that.

As they gui does not allow me to do it, I'll have to redo this locally and make a new PR later on or tomorrow..

**Comment by invidian** [comment]
@ReSearchITEng you can just use `git rebase` and `git push --force-with-lease` to amend this here. Or did you modify it via GitHub UI? If it's difficult for you, I can handle it then :)

**Comment by ReSearchITEng** [comment]
> 
> 
> @ReSearchITEng you can just use `git rebase` and `git push --force-with-lease` to amend this here. Or did you modify it via GitHub UI? If it's difficult for you, I can handle it then :)

I can't do it now, I would appreciate if you could handle it this time :)

**Review by invidian — APPROVED** [review]

**Pull request merged**